Xamarin.Forms.Platform.Android/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,284 @@
+
+		void UpdateIsEnabled() => SwipeRefreshLayout.Enabled = RefreshView.IsEnabled;
+
+		bool CanScrollUp(AView view)
+		{
+			if (!(view is ViewGroup viewGroup))
+				return base.CanChildScrollUp();
+
+			if (Forms.SdkInt < BuildVersionCodes.JellyBean)
+			{
+				if (viewGroup.IsScrollContainer)
+				{
+					return base.CanChildScrollUp();
+				}
+			}
+
+			if (!CanScrollUpViewByType(view))
+				return false;
+
+			for (int i = 0; i < viewGroup.ChildCount; i++)
+			{
+				var child = viewGroup.GetChildAt(i);
+
+				if (!CanScrollUpViewByType(child))
+					return false;
+
+				if (child is SwipeRefreshLayout)
+				{
+					return CanScrollUp(child as ViewGroup);
+				}
+			}
+
+			return true;
+		}
+
+		bool CanScrollUpViewByType(AView view)
+		{
+			if (view is AbsListView)
+			{
+				var absListView = view as AbsListView;
+				if (absListView.FirstVisiblePosition == 0)
+				{
+					var subChild = absListView.GetChildAt(0);
+
+					return subChild != null && subChild.Top != 0;
+				}
+
+				return true;
+			}
+			if(view is RecyclerView)
+			{
+				var recyclerView = view as RecyclerView;
+				return recyclerView.ScrollY < 0;
+			}
+			if (view is global::Android.Widget.ScrollView)
+			{
+				var scrollview = view as global::Android.Widget.ScrollView;
+				return scrollview.ScrollY < 0;
+			}
+			if (view is NestedScrollView)
+			{
+				var nestedScrollView = view as NestedScrollView;
+				return nestedScrollView.ScrollY < 0;
+			}
+
+			return true;
+		}
+
+		public void OnRefresh()
+		{
+			if (RefreshView?.Command?.CanExecute(RefreshView?.CommandParameter) ?? false)
+			{
+				RefreshView.Command.Execute(RefreshView?.CommandParameter);
+			}
+		}
+
+		void HandlePropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == ContentView.ContentProperty.PropertyName)
+				UpdateContent();
+			else if (e.PropertyName == VisualElement.IsEnabledProperty.PropertyName)
+				UpdateIsEnabled();
+			else if (e.PropertyName == RefreshView.IsRefreshingProperty.PropertyName)
+				UpdateIsRefreshing();
+			else if (e.IsOneOf(RefreshView.RefreshColorProperty, VisualElement.BackgroundColorProperty))
+				UpdateColors();
+
+			ElementPropertyChanged?.Invoke(sender, e);
+
+		}
+
+		public SizeRequest GetDesiredSize(int widthConstraint, int heightConstraint)
+		{
+			_renderer.View.Measure(widthConstraint, heightConstraint);
+
+			return new SizeRequest(new Size(_renderer.View.MeasuredWidth, _renderer.View.MeasuredHeight), new Size(40, 40));
+		}
+
+		public void UpdateLayout() => Tracker?.UpdateLayout();
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+				return;
+
+			if (disposing)
+			{
+				if (Element != null)
+				{
+					Element.PropertyChanged -= HandlePropertyChanged;
+				}
+
+				if (_renderer != null)
+					_renderer.View.RemoveFromParent();
+			}
+
+			_renderer?.Dispose();
+			_renderer = null;
+
+			Tracker?.Dispose();
+			Tracker = null;
+
+			_init = false;
```

Comment:
Shouldn't lines 250 - 256 be nested inside the `disposing` if?

Reply:
Right. Changed.